packages/jellyfish-network/src/BlockSubsidy.ts

```diff
@@ -0,0 +1,174 @@
+import BigNumber from 'bignumber.js'
+
+export interface CoinbaseSubsidyOptions {
+  eunosHeight: number
+  genesisBlockSubsidy: string
+  preEunosBlockSubsidy: '20000000000'
+  eunosBaseBlockSubsidy: '40504000000'
+  eunosFoundationBurn: string
+  emissionReduction: 1658
+  emissionReductionInterval: 32690
+}
+
+export const MainNetCoinbaseSubsidyOptions: CoinbaseSubsidyOptions = {
+  eunosHeight: 894000,
+  /**
+   * https://defiscan.live/blocks/279b1a87aedc7b9471d4ad4e5f12967ab6259926cd097ade188dfcf22ebfe72a
+   */
+  genesisBlockSubsidy: '59100003000000000',
+  preEunosBlockSubsidy: '20000000000',
+  eunosBaseBlockSubsidy: '40504000000',
+  /**
+   * Eunos Foundation Burn at Block Height: 894000
+   *
+   * Balances at 893999 right before it got wiped.
+   * dJEbxbfufyPF14SC93yxiquECEfq4YSd9L: 265,713,999.89000000@DFI
+   * 8UAhRuUFCyFUHEPD7qvtj8Zy2HxF5HH5nb:   2,878,893.18829048@DFI
+   */
+  eunosFoundationBurn: '26859289307829046', // 265713999.89000000 + 2,878,893.18829048
+  emissionReduction: 1658,
+  emissionReductionInterval: 32690
+}
+
+export const TestNetCoinbaseSubsidyOptions: CoinbaseSubsidyOptions = {
+  eunosHeight: 354950,
+  /**
+   * https://defiscan.live/blocks/034ac8c88a1a9b846750768c1ad6f295bc4d0dc4b9b418aee5c0ebd609be8f90?network=TestNet
+   */
+  genesisBlockSubsidy: '30400004000000000',
+  preEunosBlockSubsidy: '20000000000',
+  eunosBaseBlockSubsidy: '40504000000',
+  eunosFoundationBurn: '0',
+  emissionReduction: 1658,
+  emissionReductionInterval: 32690
+}
+
+/**
+ * All calculation are in satoshi.
+ *
+ * This class cache all 1252 reductions block subsidies and milestones allowing instantaneous computation.
+ * With very little memory footprint, 1252 x 2 BigNumber classes for both getSupply and getBlockSubsidy.
+ */
+export class BlockSubsidy {
+  private readonly reductionBlockSubsidies: BigNumber[] = this.computeBlockReductionSubsidies()
```

Comment:
Agree with the comment above, not sure how the `this` will work here, and ensure it is not compute on each instantiation.

Reply:
Compute once at init, then we can keep that object reference for future computation.

Reply:
But init on same `options` is simply wastage (if not difficult to avoid)

Reply:
Slightly confused about this statement, wastage in terms of?